src/viewport/api/project.py

```
@@ -264,6 +274,66 @@ async def get_project_detail(
+
+            archive.add(arcname=f"{folder_name}/{filename}", data=file_generator())
+
+    headers = {"Content-Disposition": f'attachment; filename="{archive_name}"'}
+    logger.info("Prepared private zip download for project %s with %s photos", project_id, total_photo_count)
+    return StreamingResponse(archive, media_type="application/zip", headers=headers)
+
+
+@router.get("/{project_id}/download/all")
+def download_project_zip(
+    project_id: uuid.UUID,
+    project_repo: ProjectRepository = Depends(get_project_repository),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)